File: problems-extend-adapt/1401-1500/1447_lowest-term-fractions/solution.ts
function lowestTermFractions(n: number): string[] {
    const gcd = (a: number, b: number): number => (b === 0 ? a : gcd(b, a % b));
    const result: string[] = [];
    for (let numer = 1; numer < n; numer++) {
        for (let denom = numer + 1; denom <= n; denom++) {
            if (gcd(numer, denom) === 1) {
                result.push(`${numer}/${denom}`);
            }
        }
    }
    return result;
}

File: problems-extend-adapt/1401-1500/1447_lowest-term-fractions/solution.rs
impl Solution {
    pub fn lowest_term_fractions(n: i32) -> Vec<String> {
        let mut result = Vec::new();
        for numer in 1..n {
            for denom in (numer + 1)..=n {
                if gcd(numer, denom) == 1 {
                    result.push(format!("{}/{}", numer, denom));
                }
            }
        }
        result
    }
}

fn gcd(a: i32, b: i32) -> i32 {
    if b == 0 {
        a
    } else {
        gcd(b, a % b)
    }
}
